### PR TITLE
fs: ext2: Remove comparing unsigned < 0 and other fixes

### PR DESCRIPTION
**Pull request by finikorg**
Remove never true comparisons for ext2, buffer overrun and de-reference before NULL check.

### CODE REVIEW

**Comment by nordicjm** [comment]
@fzdobylak

**Review by Laczen — DISMISSED** [review]

File: subsys/fs/ext2/ext2_ops.c [hunk 1]
```diff
@@ -195,7 +194,7 @@ static int ext2_lseek(struct fs_file_t *filp, off_t off, int whence)
 	}
 
 	/* New offset not inside the file. */
-	if (new_off < 0 || new_off > f->f_inode->i_size) {
+	if (new_off > f->f_inode->i_size) {
 		return -EINVAL;
 	}
```
Comment (fzdobylak) [inline subsys/fs/ext2/ext2_ops.c:199]: I am not sure if it is a proper fix. I would rather change the type of `new_off` to `int32_t` to check if resulting offset is not negative.

**Review by fzdobylak — CHANGES_REQUESTED** [review]

File: subsys/fs/ext2/ext2_ops.c (hunk 1 above)
Comment (finikorg) [reply subsys/fs/ext2/ext2_ops.c:199]: I changed it to type `off_t`

**Review by fzdobylak — APPROVED** [review]
LGTM

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**